libs/server/Storage/Session/MainStore/MainStoreOps.cs

```diff
@@ -1190,5 +1193,299 @@ public GarnetStatus MemoryUsageForKey<TContext, TObjectContext>(ArgSlice key, ou
+                    }
+
+                    var len = ComputeLCSLength(val1.ReadOnlySpan, val2.ReadOnlySpan, minMatchLen);
+                    while (!RespWriteUtils.WriteInteger(len, ref curr, end))
+                        ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+                }
+                else if (withIndices)
+                {
+                    List<LCSMatch> matches;
+                    int len;
+                    if (status1 != GarnetStatus.OK || status2 != GarnetStatus.OK)
+                    {
+                        matches = new List<LCSMatch>();
+                        len = 0;
+                    }
+                    else
+                    {
+                        matches = ComputeLCSWithIndices(val1.ReadOnlySpan, val2.ReadOnlySpan, minMatchLen, out len);
+                    }
+
+                    WriteLCSMatches(matches, withMatchLen, len, ref curr, end, ref output, ref isMemory, ref ptr, ref ptrHandle);
+                }
+                else
+                {
+                    if (status1 != GarnetStatus.OK || status2 != GarnetStatus.OK)
+                    {
+                        while (!RespWriteUtils.WriteDirect(CmdStrings.RESP_EMPTY, ref curr, end))
+                            ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+                        return GarnetStatus.OK;
+                    }
+
+                    var lcs = ComputeLCS(val1.ReadOnlySpan, val2.ReadOnlySpan, minMatchLen);
+                    while (!RespWriteUtils.WriteBulkString(lcs, ref curr, end))
+                        ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+                }
+            }
+            finally
+            {
+                if (isMemory)
+                    ptrHandle.Dispose();
+                output.Length = (int)(curr - ptr);
+            }
+
+            return GarnetStatus.OK;
+        }
+
+        private static int ComputeLCSLength(ReadOnlySpan<byte> str1, ReadOnlySpan<byte> str2, int minMatchLen)
+        {
+            var m = str1.Length;
+            var n = str2.Length;
+            var dp = new int[m + 1, n + 1];
+
+            for (int i = 1; i <= m; i++)
+            {
+                for (int j = 1; j <= n; j++)
+                {
+                    if (str1[i - 1] == str2[j - 1])
+                        dp[i, j] = dp[i - 1, j - 1] + 1;
+                    else
+                        dp[i, j] = Math.Max(dp[i - 1, j], dp[i, j - 1]);
+                }
+            }
+
+            return dp[m, n] >= minMatchLen ? dp[m, n] : 0;
+        }
+
+        private static List<LCSMatch> ComputeLCSWithIndices(ReadOnlySpan<byte> str1, ReadOnlySpan<byte> str2, int minMatchLen, out int lcsLength)
+        {
+            var m = str1.Length;
+            var n = str2.Length;
+            var dp = new int[m + 1, n + 1];
```

Comment:
Same comment here, I don't think you need the full array.

Reply:
Same comment as above :-)